### PR TITLE
Splitting Buffer Primitive classes/etc into separate files

### PR DESCRIPTION
**Pull request by ahsonkhan**
cc @KrzysztofCwalina, @shiftylogic, @davidfowl

### CODE REVIEW

**Comment by ahsonkhan** [comment]
@KrzysztofCwalina, is this PR good to go?

File: src/System.Buffers.Primitives/System/Internal/ManagedBufferPool.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 namespace System.Buffers.Internal
 {
-    internal sealed class ManagedBufferPool : BufferPool
+    internal sealed partial class ManagedBufferPool : BufferPool
```
Comment (davidfowl) [inline src/System.Buffers.Primitives/System/Internal/ManagedBufferPool.cs]: why? Move it all into a single file.

Comment (ahsonkhan) [inline src/System.Buffers.Primitives/System/Internal/ManagedBufferPool.cs]: Would the same apply to this:
https://github.com/dotnet/corefxlab/pull/1463/files#diff-82f70dec2e02480d89f3a6769347bdd9R8

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Pull request merged**